kani-compiler/src/kani_middle/stubbing/transform.rs

```diff
@@ -23,18 +26,44 @@ pub fn get_stub(tcx: TyCtxt, def_id: DefId) -> Option<DefId> {
 
 /// Returns the new body of a function/method if it has been stubbed out;
 /// otherwise, returns the old body.
-pub fn transform<'tcx>(
-    tcx: TyCtxt<'tcx>,
-    def_id: DefId,
-    old_body: &'tcx Body<'tcx>,
-) -> &'tcx Body<'tcx> {
+pub fn transform<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId, old_body: &'tcx Body<'tcx>) -> Body<'tcx> {
     if let Some(replacement) = get_stub(tcx, def_id) {
         let new_body = tcx.optimized_mir(replacement).clone();
         if check_compatibility(tcx, def_id, old_body, replacement, &new_body) {
-            return tcx.arena.alloc(new_body);
+            return new_body;
+        }
+    }
+    old_body.clone()
+}
+
+struct ExternFunctionTransformer<'tcx> {
+    tcx: TyCtxt<'tcx>,
+    callee: DefId,
+    body: Body<'tcx>,
+}
+
+impl<'tcx> MutVisitor<'tcx> for ExternFunctionTransformer<'tcx> {
+    fn tcx(&self) -> TyCtxt<'tcx> {
+        self.tcx
+    }
+
+    fn visit_operand(&mut self, operand: &mut Operand<'tcx>, _location: Location) {
+        let func_ty = operand.ty(&self.body, self.tcx);
+        if let ty::FnDef(reachable_function, generics) = *func_ty.kind() {
+            if let Some(stub) = get_stub(self.tcx, reachable_function) {
+                let Operand::Constant(fn_def) = operand else { unreachable!() };
+                fn_def.literal = ConstantKind::from_value(
+                    ConstValue::ZeroSized,
+                    self.tcx.type_of(stub).subst(self.tcx, &*generics),
+                );
+            }
         }
     }
-    old_body
+}
+
+pub fn transform_extern_functions<'tcx>(tcx: TyCtxt<'tcx>, callee: DefId, body: &mut Body<'tcx>) {
```

Comment:
maybe move this to the top (right after transform) and document it.

Reply:
Implemented as suggested.